pkg/xds/generator/egress/internal_services_generator.go

```diff
@@ -23,6 +23,11 @@ func (g *InternalServicesGenerator) Generate(
 	meshResources *core_xds.MeshResources,
 ) (*core_xds.ResourceSet, error) {
 	resources := core_xds.NewResourceSet()
+
+	if !meshResources.Mesh.ZoneEgressEnabled() {
+		return resources, nil
```

Comment:
because in egress proxy builder I had to change `ZoneEgressEnabled` to just `MTLSEnabled` I had to add a condition here for only internal services so that we won't waste CP resources on reconfiguring egress when it's not used